core/controllers/editor.py

```diff
@@ -854,14 +858,22 @@ def post(self, exploration_id):
                 'Expected a filename ending in .%s, received %s' %
                 (file_format, filename))
 
-        # Save the file.
-        fs = fs_domain.AbstractFileSystem(
-            fs_domain.ExplorationFileSystem(exploration_id))
-        if fs.isfile(filename):
+        mimetype = 'image/' + str(extension)
+
+        # Image files are stored to the datastore in the dev env, and to GCS
+        # in production.
+        file_system_class = (
+            fs_domain.ExplorationFileSystem if feconf.DEV_MODE
+            else fs_domain.GcsFileSystem)
+        fs = fs_domain.AbstractFileSystem(file_system_class(exploration_id))
+        filepath = 'image/' + filename
```

Comment:
Why do you construct this both here and on line 875? Use self._FILENAME_PREFIX as well.